libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -80,6 +80,25 @@ void StaticAnalyzer::endVisit(FunctionDefinition const&)
 	m_localVarUseCount.clear();
 }
 
+bool StaticAnalyzer::visit(ModifierInvocation const& _modifier)
+{
+	if (m_constructor)
+		if (auto decl = dynamic_cast<ContractDefinition const*>(_modifier.name()->annotation().referencedDeclaration))
+			for (auto const& specifier: m_currentContract->baseContracts())
+				if (specifier->name().annotation().referencedDeclaration == decl && !specifier->arguments().empty())
+				{
+					SecondarySourceLocation ssl;
+					ssl.append("Overriden constructor call is here:", specifier->location());
```

Comment:
Should use `ssl.limitSize()` from #3310 here.

Reply:
I am not appending in a loop, so I see no reason for using `limitSize`.

Reply:
Actually you should list all shadowed ones if possible.

Reply:
It's clearly an error if we shadow the super constructor within the same contract, but should we enforce it across the whole inheritance hierarchy? I haven't really thought much about it, so I opted for providing flexibility.